data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TargetsParameterConfig.java

```diff
@@ -0,0 +1,137 @@
+package org.opensearch.dataprepper.plugins.processor.translate;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.AssertTrue;
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.NotNull;
+import org.apache.commons.lang3.Range;
+import org.opensearch.dataprepper.plugins.processor.mutateevent.TargetType;
+import org.opensearch.dataprepper.typeconverter.TypeConverter;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+public class TargetsParameterConfig {
+    private final TypeConverter converter;
+    private final LinkedHashMap<Range<Float>, Object> rangeMappings = new LinkedHashMap<>();
+    private final Map<String, Object> individualMappings = new HashMap<>();
+    private final Map<Pattern, Object> compiledPatterns = new HashMap<>();
+    @JsonProperty("target")
+    @NotNull
+    @NotEmpty
+    private String target;
```

Comment:
If this is under `targets`, why not just name this `key`? `key` is a more accurate description of what is being configured here

Reply:
Yes there could be change in the naming for `target` option and `target_type` option. target can be changed to `key` or `name` and `target_type` will be re-named to `type`